arelle/PackageManager.py

```diff
@@ -663,3 +759,60 @@ def removePackageModule(cntlr, name):
         global packagesConfigChanged
         packagesConfigChanged = True
     return result
+
+invalidZipDirEntryPattern = re.compile(r"^/|^.*\\")
+forbiddenDirEntryPattern = re.compile(r"^..?$|^..?/|^.*/..?/|^.*//")
+def validatePackageEntries(filesource, errors=None):
+    result = False
+    if filesource is not None and filesource.isArchive:
+        filesource.setLogErrors(errors)
+        pathCounts = {}
+        try:
+            _dir = filesource.dir
+        except zipfile.BadZipFile:
+            filesource.setLogErrors(None)
+            return False
+        for f in _dir:
+            pathCounts[f] = pathCounts.get(f,0) + 1
+        if not _dir:
+            filesource.cntlr.addToLog(_("Archive has no files"), 
+                                      messageCode="rpe:invalidDirectoryStructure",
+                                      level=logging.ERROR,
+                                      file=filesource.url)
+            if errors is not None: errors.append("rpe:invalidDirectoryStructure")
+        elif any (invalidZipDirEntryPattern.match(f) for f in _dir):
+            for f in filesource.dir:
+                if invalidZipDirEntryPattern.match(f):
+                    filesource.cntlr.addToLog(_("Archive must not contain absolute path references or backslashes \"%(name)s\""), 
+                                              messageCode="rpe:invalidArchiveFormat",
+                                              messageArgs={"name":f},
+                                              level=logging.ERROR,
+                                              file=filesource.url)
+                    if errors is not None: errors.append("rpe:invalidArchiveFormat")
+        elif any (forbiddenDirEntryPattern.match(f) for f in _dir):
+            for f in filesource.dir:
+                if forbiddenDirEntryPattern.match(f):
+                    filesource.cntlr.addToLog(_("Archive contains forbidden file name \"%(name)s\""), 
+                                              messageCode="rpe:invalidDirectoryStructure",
+                                              messageArgs={"name":f},
+                                              level=logging.ERROR,
+                                              file=filesource.url)
+                    if errors is not None: errors.append("rpe:invalidDirectoryStructure")
+        elif sum("reportPackage.json" in f for f in _dir) > 1:
```

Comment:
This check seems too loose.  It seems like `x/reports/my-reportPackage.json`, `x/reports/reportPackage.json.xml`, `x/reportPackage.json/report.json` are all valid entries that should not be considered as duplicate `x/META-INF/reportPackage.json` files.